protocol/messenger.go

```diff
@@ -78,6 +78,7 @@ type Messenger struct {
 	installationID             string
 	mailserver                 []byte
 	database                   *sql.DB
+	quit                       chan struct{}
```

Comment:
Why is this a `struct{]` and not a bool for example?

Does the `quit` field have a write event? I can't see it?

Reply:
not sure, I always seen that being used, I think it's because of memory https://stackoverflow.com/questions/52035390/why-using-chan-struct-when-wait-something-done-not-chan-interface

Reply:
it's closed in `Shutdown()` https://github.com/status-im/status-go/pull/2022/files#diff-5df3979bc10f2ff16f7942ec557947ecR570

Reply:
That's interesting. It feels hacky, but I see why one would use `struct{}`.